src/app/client/src/app/plugins/location/components/location-selection/location-selection.component.ts

```diff
@@ -119,7 +120,29 @@ export class LocationSelectionComponent implements OnInit, OnDestroy, AfterViewI
       }
     } else {
       const result: any = await this.sbFormLocationSelectionDelegate.formGroup;
-      this.registerSubmit.emit(_.get(result, 'value'));
+      const locationDetails: SbLocation[] = Object.keys(_.get(result, 'value.children.persona'))
+        .reduce<SbLocation[]>((acc, key) => {
+          const locationDetail: SbLocation | null = _.get(result, 'value.children.persona')[key];
+          if (_.get(locationDetail, 'code')) {
+            acc.push(locationDetail);
+          }
+          return acc;
+        }, []);
+      const userTypes = [{ type: 'teacher' }];
+      const payload: any = {
+        userId: _.get(this.userService, 'userid'),
+        profileLocation: locationDetails,
+        profileUserTypes: userTypes
+      };
+      this.locationService.updateProfile(payload).toPromise()
+        .then((res) => {
+          console.log('res ', res); // TODO: log!
```

Comment:
do we need console.log

Reply:
@rajeevsathish since this workflow cannot be tested in local environment; post testing in staging, log statements will be removed.